### ISSUE TITLE
detect unnecessary curly braces

### ISSUE THREAD

**Issue by mimoo** [rust-lang/rust-clippy#9974]
### What it does

In the following example, there's a single expression inside curly braces. I think clippy should detect that because it is probably not intended. (We have some of these in our codebase due to refactor.)

### Lint Name

unecessary_curly_braces

### Category

style

### Advantage

- Remove lines of code
- Removes indentation
- (Improves readability)

### Example

```rust
let a = {
  if cond {
    None
  } else {
    Some(5)
  }
};
```

Better:

```rust
let a = if cond {
  None
} else {
  Some(5)
};
```
